### ISSUE TITLE
wrapping streams does not work

### ISSUE THREAD

**Issue by hij1nx** [dominictarr/rpc-stream#16]
In the readme it says that this should work, but it doesn't...

```
var rpc = require('rpc-stream')
var net = require('net')

net.createServer(function(con) {
  var server = rpc(require('fs'))
  server.pipe(con).pipe(server)
}).listen(3000, function() {

  var client = rpc()
  var con = net.connect(3000)
  client.pipe(con).pipe(client)

  var remote = client.wrap(require('fs'))
  console.log(remote.createReadStream('./a.js'))
})
```

**Comment by dominictarr** [comment]
hmm, what output do you get?

**Comment by hij1nx** [comment]
"undefined"

**Comment by hij1nx** [comment]
eh, actually, I think it might return `[Function]` or `undefined`, can't remember at the moment.

**Comment by dominictarr** [comment]
Oh sorry I didn't process this issue properly - the probem here is that I wrote rpc-stream before I thought that having streams within rpc would be a good idea... fun fact: I just wrote this to troll substack into exposing a stream instead of wrapping a server. That was highly successful, but I never actually used it for anything until we where optimizing multilevel.

Multilevel got streams + cbs by running rpc-stream inside mux-demux, and then using the other mux-demux streams for everything else...
More recently I realized that most node apis have a mix of streams and async, and that the core thing in both multiplexing and rpc protocols is framing: you are putting a bunch of items down a socket, and need some way to delimit them... so if the hard part is framing, can we just have a multiplexer that also did rpc?

The answer to that question is [muxrpc](https://github.com/ssbc/muxrpc).
